src/datastore/postgres-store.ts

```diff
@@ -1712,7 +1712,7 @@ export class PgDataStore
         `
         SELECT MAX(nonce) nonce
         FROM txs
-        WHERE sender_address = $1
+        WHERE ((sender_address = $1 AND sponsored = false) OR (sponsor_address = $1 AND sponsored= true))
```

Comment:
These sql query changes can have considerable performance impact. Previously these were simple equality conditions against an index column `sender_address`. Now, it's checking multiple columns, and I don't think the `sponsor_address` or `sponsored` columns are indexed, and there would also have to be composite index now for both pairs of columns.

Reply:
@asimm241 do you have a local pg db with these changes and the new indexes? I'd like to see the performance impact, and the `EXPLAIN` result to show that the indexes are used for this query.

Reply:
I ran this query on a local pg instance with full prod data. Performance difference seems alright, although hard to compare that to how it will perform on google cloud sql.

@rafaelcr can you also give this a review?